### ISSUE TITLE
Private site redirects to /undefined when visiting direct to `/private/`

### ISSUE THREAD

**Issue by allouis** [TryGhost/Ghost#9959]
To replicate:

Create private blog

visit privateblog.com/private/

**Comment by kirrg001** [comment]
Wasn't this fixed by https://github.com/TryGhost/Ghost/commit/9f6f01d44aae66060b2f5f0b9a24eeb2b14bcb8c? 🤔 

I am wondering why currently bugs pop up for private blogging.
Is it related to https://github.com/TryGhost/Ghost/commit/045ce0a800e05d2a5093de6285d5c4364128570e somehow?

**Comment by allouis** [comment]
It was, but not for if you visited `/private/` with a trailing slash. I will have a PR to fix it in a sec

**Comment by kirrg001** [comment]
@allouis Okay 👍

> I am wondering why currently bugs pop up for private blogging.

?

**Comment by allouis** [comment]
To elaborate on why this wasn't fixed with https://github.com/TryGhost/Ghost/commit/9f6f01d44aae66060b2f5f0b9a24eeb2b14bcb8c It was because of an error I made in https://github.com/TryGhost/Ghost/commit/045ce0a800e05d2a5093de6285d5c4364128570e when testing to see if the redirect param was missing. Instead of testing for the param, i tested for the query object - which always exists. So `undefined` was being passed into the redirect. The reason I didn't catch this when testing is because when I visited `/private` the url cleaner would redirect me to `/private/` and so a query param would be attached!
